src/components/molecules/event/myEventCard/myEventCard.tsx

```diff
@@ -0,0 +1,284 @@
+import React, { useContext, useEffect, useState } from 'react';
+import styles from './myEventCard.module.scss';
+import { Event, EventOptions, JoinEventPayload } from 'models/apiModels';
+import EventHeader from '../eventHeader/EventHeader';
+import Icon from 'components/atoms/icons/icon';
+import { Link } from 'react-router-dom';
+import {
+  getEventOptions,
+  updateEventOptions,
+  isConfirmedEvent,
+  getJoinedParticipants,
+  getMemberAssociatedWithToken,
+} from 'api';
+import { AuthenticateContext } from 'contexts/authProvider';
+import Button from 'components/atoms/button/Button';
+import EventPreferences from '../eventPreferences/EventPreferences';
+import useModal from 'hooks/useModal';
+import Modal from 'components/molecules/modal/Modal';
+import { useToast } from 'hooks/useToast';
+import { transformDate } from 'utils/timeConverter';
+import classNames from 'classnames';
+
+const NA = () => {
+  return <span className={styles.NA}>N/A</span>;
+};
+
+interface SpotStatus {
+  message: string;
+  color: string;
+}
+
+/* Pill component to display confirmation status */
+const StatusPill: React.FC<SpotStatus> = ({ message, color }) => {
+  const classes = classNames(`${styles[color]}`, styles.status);
+  return <div className={classes}>{message}</div>;
+};
+
+export interface MyEventCardProps {
+  eventData: Event;
+}
+
+const MyEventCard: React.FC<MyEventCardProps> = ({ eventData }) => {
+  /* Options pulled from backend */
+  const [options, setOptions] = useState<EventOptions>({
+    food: false,
+    transportation: false,
+    dietaryRestrictions: '',
+  });
+  /* Payload to send */
+  const [editPrefsPayload, setEditPrefsPayload] = useState<JoinEventPayload>({
```

Comment:
I thiiink we can remove one of these states and work with just one of them🤔

Reply:
My motivation for having two states here is to have the options displayed on the event card only update once the change has been made and 200 has been returned from API. With just one state, the options would change immediately on form toggle and be inconsistent with actual options.

Reply:
Yeah okay that makes sense🙌